lib/Controller/PasswordController.php

```diff
@@ -93,6 +93,13 @@ protected function createPasswordTemplateResponse($redirect_url, $error = null)
 			'redirect_url' => $redirect_url,
 			'webroot' => \OC::$WEBROOT
 		];
+
+		$user = $this->userSession->getUser();
+		if ($user !== null) {
+			if ($this->config->getUserValue($user->getUID(), 'password_policy', 'firstLoginPasswordChange') === '1') {
```

Comment:
another approach would be the reverse: set a flag after the first login password was set. but ok, this works too